src/CMakeLists.txt

```diff
@@ -326,13 +326,18 @@ if(NOT TARGET RapidJSON::RapidJSON)
 endif()
 
 option(WITH_FMT_HEADER_ONLY "use header-only version of fmt library" OFF)
-set(WITH_FMT_VERSION "8.1.1" CACHE
-  STRING "build with fmt version")
-find_package(fmt ${WITH_FMT_VERSION} QUIET)
+option(WITH_SYSTEM_FMT "build with system fmt" NO)
```

Comment:
per the commit message:

> Add `WITH_SYSTEM_FMT` option to request use of system fmt and fail if
> requested and not found. Remove the Findfmt.cmake build module and use
> the config file provided with fmt. Require 9.1 ≤ version < 10.
> 

there are following changes

1. Add `WITH_SYSTEM_FMT` option to request use of system fmt and fail if requested and not found. 
2. Remove the Findfmt.cmake build module and use the config file provided with fmt. 
3. Require 9.1 ≤ version < 10.

the changes in this commit match the commit message, so great!

but the questions remain. why, why and why. and why


1. why add `WITH_SYSTEM_FMT` ? what problem does it address?
2. why remove `Findfmt.cmake`?
3. and why require Require 9.1 ≤ version < 10 ? and why is it safe to do so? do distros that we care about include the required fmtlib-dev / fmt-devel / ... ?
4. are these change depending on each other so we cannot split them into smaller ones?

Reply:
[snip]
> but the questions remain. why, why and why. and why
> 
>     1. why add `WITH_SYSTEM_FMT` ? what problem does it address?

Previously, there was no way to opt out of using the system fmt if you didn't want to. The immediate problem was that whenever your distribution upgraded to a newer fmt, Ceph builds would fail. This was very vexing when we had a a submodule right there.

>     2. why remove `Findfmt.cmake`?

It ignored the version string and would successfully 'find' fmt whatever the version requirements were.
 
>     3. and why require Require 9.1 ≤ version < 10 ? and why is it safe to do so? do distros that we care about include the required fmtlib-dev / fmt-devel / ... ?

It's the version our submodule uses and when I proposed doing this in ceph-devel people asked me to take out the compatibility code for versions of fmt below 9. However looking at it jammy doesn't actually have a version about 8.x. So I'll change the version back to 8.1 and remove the second commit.

>     4. are these change depending on each other so we cannot split them into smaller ones?

We can split adding WITH_SYSTEM_FMT and the part that makes version checks work apart. I shall do that now.